libraries/WebServer/src/Parsing.cpp

```diff
@@ -304,12 +304,39 @@ void WebServer::_uploadWriteByte(uint8_t b){
 
 int WebServer::_uploadReadByte(WiFiClient& client){
   int res = client.read();
-  if(res == -1){
-    while(!client.available() && client.connected())
-      delay(2);
-    res = client.read();
+  if(res < 0) {
+    // keep trying until you either read a valid byte or timeout
+    const unsigned long startMillis = millis();
```

Comment:
why const?

Reply:
> why const?

because it is only written to once in the context of this function... may as well declare it as such

Reply:
only written to once in this function, so I think it's correct to declare it as such.

Reply:
but you set it every time you call this function....

Reply:
removed the const keywwords